core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -155,51 +210,74 @@ object ConsumerGroupCommand {
         deleteAllForTopic()
     }
 
-    protected def describeGroup(group: String) {
+    protected def collectGroupAssignment(group: String): (Option[String], Option[Seq[PartitionAssignmentState]]) = {
       val props = if (opts.options.has(opts.commandConfigOpt)) Utils.loadProps(opts.options.valueOf(opts.commandConfigOpt)) else new Properties()
       val channelSocketTimeoutMs = props.getProperty("channelSocketTimeoutMs", "600").toInt
       val channelRetryBackoffMs = props.getProperty("channelRetryBackoffMsOpt", "300").toInt
+      if (!zkUtils.getConsumerGroups().contains(group))
+        return (None, None)
+
       val topics = zkUtils.getTopicsByConsumerGroup(group)
-      if (topics.isEmpty)
-        println("No topic available for consumer group provided")
-      printDescribeHeader()
-      topics.foreach(topic => describeTopic(group, topic, channelSocketTimeoutMs, channelRetryBackoffMs))
-    }
+      val topicPartitions = getAllTopicPartitions(topics)
+      var groupMemberIds = zkUtils.getConsumersInGroup(group)
+
+      // mapping of topic partition -> consumer id
+      val consumerIdByTopicPartition = topicPartitions.map { topicPartition =>
+        val owner = zkUtils.readDataMaybeNull(new ZKGroupTopicDirs(group, topicPartition.topic).consumerOwnerDir + "/" + topicPartition.partition)._1.get
+        // cut the appending thread id
+        val splitPos = owner.lastIndexOf('-')
+        topicPartition -> owner.substring(0, splitPos)
+      }.toMap
+
+      // mapping of member id -> list of topic partitions
+      val memberTopicPartitions = consumerIdByTopicPartition groupBy{_._2} map {
+        case (key, value) => (key, value.unzip._1.toArray) }
 
-    private def describeTopic(group: String,
-                              topic: String,
-                              channelSocketTimeoutMs: Int,
-                              channelRetryBackoffMs: Int) {
-      val topicPartitions = getTopicPartitions(topic)
-      val groupDirs = new ZKGroupTopicDirs(group, topic)
-      val ownerByTopicPartition = topicPartitions.flatMap { topicPartition =>
-        zkUtils.readDataMaybeNull(groupDirs.consumerOwnerDir + "/" + topicPartition.partition)._1.map { owner =>
-          topicPartition -> owner
+      // mapping of member id -> list of subscribed topics
+      val topicsByMemberId = zkUtils.getTopicsPerMemberId(group)
+
+      var assignmentRows = topicPartitions.flatMap { topicPartition =>
+        val partitionOffsets = getPartitionOffsets(group, List(topicPartition), channelSocketTimeoutMs, channelRetryBackoffMs)
+        val memberId = consumerIdByTopicPartition.get(topicPartition)
+        // since member id is repeated in client id, leave member host and client id empty
+        if (!memberId.isEmpty) {
```

Comment:
Yeah, something like that. Kind of hard to tell sometimes when you're trying to get a little too cute, but using `map` and `foreach` seems to generally be preferred over an explicit check for `isDefined` or `isEmpty`.